src/Streamly/Internal/Data/List.hs

```diff
@@ -6,6 +6,10 @@
 {-# LANGUAGE UndecidableInstances       #-} -- XXX
 {-# LANGUAGE ViewPatterns               #-}
 
+#if MIN_VERSION_base(4,17,0)
```

Comment:
Why is this required here?

Reply:
To satisfy `-Wtype-equality-requires-operators`:

```
$ cabal build all --enable-tests --enable-benchmarks --project-file=cabal.upgrade.project
Build profile: -w ghc-9.4.5 -O1
...
[42 of 68] Compiling Streamly.Internal.Data.List

src/Streamly/Internal/Data/List.hs:117:13: warning: [-Wtype-equality-requires-operators]
    The use of ‘~’ without TypeOperators
    will become an error in a future GHC release.
    Suggested fix: Perhaps you intended to use TypeOperators
    |
117 | instance (a ~ Char) => IsString (List a) where
    |             ^

src/Streamly/Internal/Data/List.hs:179:13: warning: [-Wtype-equality-requires-operators]
    The use of ‘~’ without TypeOperators
    will become an error in a future GHC release.
    Suggested fix: Perhaps you intended to use TypeOperators
    |
179 | instance (a ~ Char) => IsString (ZipList a) where
    |             ^
 ```
 
 With earlier versions of GHC, this is not needed, hence the `CPP` conditional. The other option is not to `CPP` guard this language extension as `-XTypeOperators` has been available since `ghc-6.8.1`.

Reply:
I removed the `CPP` guard but left the `-XTypeOperators` extension.